pkg/provider/azure_loadbalancer_test.go

```diff
@@ -3785,6 +3785,52 @@ func TestReconcileSecurityGroup(t *testing.T) {
 				},
 			},
 		},
+		{
```

Comment:
Do we have a test case where the sg rule should be deleted completely?

Reply:
Please refer to https://github.com/kubernetes-sigs/cloud-provider-azure/blob/3f6d95e591a14322599495c6ffd30f68f6aa7d1c/tests/e2e/network/network_security_group.go#L143-L191